src/primitives/poseidon.rs

```diff
@@ -300,7 +298,7 @@ impl<F: FieldExt, S: Spec<F, T, RATE>, const T: usize, const RATE: usize, const
 pub struct Hash<
     F: FieldExt,
     S: Spec<F, T, RATE>,
-    D: Domain<F, S, T, RATE>,
+    D: Domain<F, T, RATE>,
```

Comment:
I looked back at the history to see why this relationship was present. The reason is that I originally implemented this primitive before we had const generics, so `Domain` was using the old associated types `S::State` and `S::Rate`. Now that we are using const generics, the consistent parameterization of `Spec` and `Domain` is enforced here, in the type signature of `Hash`.